extensions/resteasy-reactive/quarkus-resteasy-reactive-links/runtime/src/main/java/io/quarkus/resteasy/reactive/links/runtime/GetterAccessorsContainer.java

```diff
@@ -11,7 +11,12 @@ public class GetterAccessorsContainer {
     private final Map<String, Map<String, GetterAccessor>> getterAccessors = new HashMap<>();
 
     public GetterAccessor get(String className, String fieldName) {
-        return getterAccessors.get(className).get(fieldName);
+        try {
+            return getterAccessors.get(className).get(fieldName);
+        } catch (NullPointerException ex) {
+            throw new IllegalStateException("Cannot generate web links for " + className
+                    + " because the class does not contain an `id` field");
```

Comment:
Shouldn't it be:
```suggestion
                    + " because the class does not contain a field named `" + fieldName + "`");
```
?
It looks more future-proof.

Reply:
But the issue if not `fieldName` missing, is that is it always expected a field `id` to be present. 
I don't know who came up with the rules, but that's the current implementation.

Reply:
OK, so I'm the one who is confused :).

I think what confuses me is that catching the exception here is a bit of an anti-pattern. Is there any way to catch this condition earlier when inspecting the class instead of that late when getting an unrelated field name?

Note that I haven't looked at the code at all so I'm giving this advice just by inspecting the diff. I might be completely wrong :).

Reply:
I tried to catch this in the deployment module, in `LinksProcessor.implementPathParameterValueGetters`, after setting the data in `getterAccessorsContainer`.
I tried it by adding a `contains` method to the `GetterAccessorsContainer` and checking if it contains the fieldname `id`, but I got an error  because it involves calling a `getValue`: `You cannot invoke getValue() directly on an object returned from the bytecode recorder, you can only pass it back into the recorder as a parameter`.
I'm totally open for a better implementation. I'm also not very fond of this one

Reply:
Hey, sorry for the late reply, I was on PTO.

Yes, so this all boils down to Quarkus architecture. The build steps are executed at build time and you can record (via a recorder) things that will be executed at runtime (either at static init or runtime init). A value that is initialized by a recorder (the `RuntimeValue` you have here) is a pointer to what you will obtain later at runtime. You can't do anything with it in the build steps... but you can pass it to a recorder method.

So I would recommend you add a new method to the recorder to validate things and call it at the end of `implementPathParameterValueGetters`. This is very similar to what is already done in this method (typically with `getterAccessorsContainerRecorder.addAccessor` for instance).

Let me know if something is unclear, I will be happy to help.